src/main.py

```diff
@@ -57,18 +57,21 @@ def __init__(self, app_name: str = None, app_version: str = None, tid: str = Non
             self.backend.generate_new_uid_file()
 
         # Consent file may be absent, for example, during the first run of Openvino tool.
-        # In this case we trigger opt-in dialog that asks user permission for sending telemetry.
         if opt_in_check_result == ConsentCheckResult.NO_FILE:
-            if opt_in_checker.create_or_check_consent_dir():
-                answer = ConsentCheckResult.DECLINED
+            # Try to create empty directory for consent file
+            if not opt_in_checker.create_or_check_consent_dir():
+                return
+            if backend == 'ga':
 
                 # check if it is openvino tool
                 if not self.check_by_cmd_line_if_dialog_needed():
                     return
 
-                # create openvino_telemetry file if possible with "0" value
+                # create consent file if possible with "0" value
+                answer = ConsentCheckResult.DECLINED
                 if not opt_in_checker.update_result(answer):
                     return
+                # For GA backend we trigger opt-in dialog that asks user permission for sending telemetry.
```

Comment:
will we have the same dialog for Matomo?

Reply:
I thought that the logic below should be utilized for matomo

Reply:
No, the main reason why we want to switch to Matomo is the possibility to turn on telemetry by default. So should only inform a user that telemetry is collected and provide the instruction how to turn it off.
Currently Soren double checks it with legal.